### DISCUSSION TITLE
Prevent proliferation of c source files when using automatic code generation (JIT compiling)

### DISCUSSION THREAD

**Discussion by billtubbs** [casadi/casadi#3721] [Q&A]
I'm building on one of the example scripts ([sys_id.py](https://github.com/casadi/casadi/blob/main/docs/examples/python/sysid.py)) which involves automatic code generation and I noticed a bunch of c source files appearing in my file system:

```
jit_tmp0UYLz5.c
jit_tmp1Eb7dA.c
jit_tmp5FyemE.c
jit_tmpbwqLsG.c
jit_tmpiI4jiJ.c
jit_tmpKbgjsI.c
jit_tmpKWr2U4.c
jit_tmpLHuasM.c
jit_tmpNjBoSy.c
jit_tmpOIJaRt.c
```

It seems to generate new files every time I run the script, and some of these are 15 MB in size...

Is there a way to delete redundant files or overwrite old versions so they don't proliferate?

**Comment by jgillis** [comment]
By default, these should get deleted. What OS are you using?

**Reply by billtubbs** [reply]
I'm using WSL (Windows Subsystem for Linux) in a Docker container.  You're right it is OS specific.  When I run the file on my other Mac OS X computer it doesn't happen.  Anything I can do to help debug?
